app/models/article.rb

```diff
@@ -1,5 +1,22 @@
 class Article < ApplicationRecord
+  #validations
   validates :title, presence: true
   validates :body, presence: true, length: {minimum:10, to_short:"minimum is %{count} characters"}
-  validates :author, presence: true, uniqueness: true
+  validates :author_id, presence: true
+  validates :published, inclusion: [true, false]
+
+  #association
+  belongs_to :author # An article has one author hence it belongs to a single author
+
+  #callback
+  before_save ->{ # if the author is not a premium user, then he cannot publish more than five articles
+    author=Author.find(self.author_id)
+    if !author.premium_user # checking whether author is a premium user or not.
+      articles=Article.where(author_id: author.id, published: true)
+      if(articles.length>=5)
```

Comment:
use count instead of lenght

Reply:
changes implemented, please check latest commit.